src/main/java/com/aws/greengrass/util/platforms/windows/WindowsPlatform.java

```diff
@@ -121,9 +128,28 @@ public void addUserToGroup(String user, String group) throws IOException {
     }
 
     @Override
-    protected void setPermissions(FileSystemPermission permission, Path path,
-                                  EnumSet<FileSystemPermission.Option> options) throws IOException {
-        // [P41372857]: Implement using ACL for Windows
+    protected void setOwner(FileSystemPermission permission, Path path) throws IOException {
+        UserPrincipalLookupService lookupService = path.getFileSystem().getUserPrincipalLookupService();
+        UserPrincipal userPrincipal = lookupService.lookupPrincipalByName(permission.getOwnerUser());
+        FileOwnerAttributeView view = Files.getFileAttributeView(path, FileOwnerAttributeView.class,
+                LinkOption.NOFOLLOW_LINKS);
+
+        logger.atTrace().setEventType(SET_PERMISSIONS_EVENT).kv(PATH, path).kv("owner", permission.getOwnerUser())
+                .log();
+        view.setOwner(userPrincipal);
+
+        // Note that group ownership is not used.
+    }
+
+    @Override
+    protected void setMode(FileSystemPermission permission, Path path) throws IOException {
+        List<AclEntry> acl = permission.toAclEntries(path);
+        logger.atTrace().setEventType(SET_PERMISSIONS_EVENT).kv(PATH, path).kv("perm", acl.toString()).log();
+        AclFileAttributeView view = Files.getFileAttributeView(path, AclFileAttributeView.class,
+                LinkOption.NOFOLLOW_LINKS);
+        List<AclEntry> existingAcl = view.getAcl();
```

Comment:
we need to rewrite the existing acl. For example, if someone allowed everyone to read their private key, then we want to get rid of that